pkg/start/start.go

```diff
@@ -204,46 +192,19 @@ func (o *Options) run(ctx context.Context, controllerCtx *Context, lock *resourc
 		}()
 	}
 
-	exit := make(chan struct{})
-	exitClose := sync.Once{}
-
-	// TODO: when we switch to graceful lock shutdown, this can be
-	// moved back inside RunOrDie
-	// TODO: properly wire ctx here
-	go leaderelection.RunOrDie(context.TODO(), leaderelection.LeaderElectionConfig{
-		Lock:          lock,
-		LeaseDuration: leaseDuration,
-		RenewDeadline: renewDeadline,
-		RetryPeriod:   retryPeriod,
+	go leaderelection.RunOrDie(runContext, leaderelection.LeaderElectionConfig{
+		Lock:            lock,
+		ReleaseOnCancel: true,
+		LeaseDuration:   leaseDuration,
+		RenewDeadline:   renewDeadline,
+		RetryPeriod:     retryPeriod,
 		Callbacks: leaderelection.LeaderCallbacks{
 			OnStartedLeading: func(localCtx context.Context) {
 				controllerCtx.Start(runContext)
```

Comment:
`runContext` -> `localCtx` ? i'm hoping that RunOrDie passes on the child context from the passed ctx `runContext` to the OnStartedLeading function..

Reply:
> `runContext` -> `localCtx` ?

I've pivoted the local helper here to just use `ctx` with 621d0cd1 -> 235116dd, so now I'm no longer reaching around and grabbing an external context.